### PR TITLE
OSW-2067: Add E2E tests for plots page

### PR DESCRIPTION
**Pull request by nemovc**
Add some frontend end-to-end tests

First of all, we set up/configure playwright to run end to end tests. Since there are already some playwright tests (stress tests) the new tests are in a different configuration file (`playwright.mocked.config.js`) 

Then we add some helper functions to generate fixture data, as well as some default fixtures stored in the repo itself (these fixture generators are designed to be as flexible as possible; there's also a helper to allow you to persist it to disk for further modification, but most can be generated on-the-fly in each test)

Then we add some e2e tests for the plots page, checking that it renders; that the various controls (plot visibility, plot format, time range selection bar) work, and that click and drag selection works (this is a bit janky and fragile, but does technically work)

Finally we set up jenkins CI  to also run the e2e tests on push (@sebastian-aranda

### CODE REVIEW

File: tests/e2e/helpers/constants.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// @ts-check
+
+// Fixed dayobs matching the fixture data (LSSTCam night).
+// On localhost the retention policy is null, so any past date is valid.
+export const TEST_DAYOBS = "20260101";
+export const TEST_DAYOBS_INT = 20260101;
+export const PLOTS_URL = `/nightlydigest/plots?startDayobs=${TEST_DAYOBS}&endDayobs=${TEST_DAYOBS}&telescope=Simonyi`;
+
+// Full time range UTC boundaries for dayobs=20260101:
```
Comment (emanehab99) [inline tests/e2e/helpers/constants.js:7]: Are these comments still needed?

File: tests/e2e/helpers/mock-api.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+const DEFAULT_MOCKS = {
+  "data-log": loadFixture("data-log"),
+  almanac: loadFixture("almanac"),
+  version: { version: "test" },
+};
+
+/**
+ * Sets up Playwright route mocks for all backend API endpoints used by the
+ * Plots page. Call this in beforeEach before navigating.
```
Comment (emanehab99) [inline tests/e2e/helpers/mock-api.js]: Will this function be used to mock all backend API endpoints, or just the ones used by the Plots page?

File: tests/e2e/mocked/plots/almanac.spec.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+// Timeline:
+//  night start │ twilight_evening │ moon_rise │ moon_set │ twilight_morning │ night end
+//  12:00 UTC   │ 20:00 UTC        │ 22:00 UTC │ 02:00 UTC│ 04:00 UTC        │ 12:00 UTC
+//
+// TAI equivalents (UTC ms + 37 000):
+const TWILIGHT_EVENING_TAI = new Date("2026-01-01T20:00:00Z").getTime() + 37000; // 1767297637000
+const TWILIGHT_MORNING_TAI = new Date("2026-01-02T04:00:00Z").getTime() + 37000; // 1767326437000
+const MOON_RISE_TAI = new Date("2026-01-01T22:00:00Z").getTime() + 37000; // 1767304837000
+
+const ALMANAC_MOCK = {
```
Comment (emanehab99) [inline tests/e2e/mocked/plots/almanac.spec.js]: Is there a reason for not using the almanac fixture?

File: tests/e2e/mocked/plots/format.spec.js [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+    await page.locator("label[for='band-bandColor']").click();
+    await page.keyboard.press("Escape");
+
+    const airmassChart = page.locator('[data-slot="chart"][title="Airmass"]');
+    // Airmass has bandMarker=false so its dots must NOT be coloured by band
+    await expect(
+      airmassChart.locator('[data-obsid][fill="#fdc900"]'),
+    ).toHaveCount(0);
+  });
+});
```
Comment (emanehab99) [inline tests/e2e/mocked/plots/format.spec.js:308]: One small addition to consider here is a test for the `Show X Axis Label` toggle

**Review by emanehab99 — APPROVED** [review]
This looks great, Owen! 
Really nice coverage  overall, especially around real user interactions and edge case handling.
The helper extraction makes the suite easy to read and gives the spec files a nice behavior-focused structure.

One thing I noticed is that fixtures and tests cover single night behavior. I'd consider covering multi-night behavior in the timeline and the plot tests; covering the zigzag twilight lines, and the combination of no-exposure nights and normal nights.

I've also left a couple of comments/questions for your consideration.

File: tests/e2e/helpers/mock-api.js (hunk 2 above)
Comment (nemovc) [reply tests/e2e/helpers/mock-api.js]: Good catch - it should be for all pages, its just that Plots was the first one implemented

File: tests/e2e/mocked/plots/almanac.spec.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+// Almanac events placed well within the observable night.
+// UTC strings are converted internally via UTC + 37 s → TAI ms.
+//
+// Timeline:
+//  night start │ twilight_evening │ moon_rise │ moon_set │ twilight_morning │ night end
+//  12:00 UTC   │ 20:00 UTC        │ 22:00 UTC │ 02:00 UTC│ 04:00 UTC        │ 12:00 UTC
+//
+// TAI equivalents (UTC ms + 37 000):
+const TWILIGHT_EVENING_TAI = new Date("2026-01-01T20:00:00Z").getTime() + 37000; // 1767297637000
```
Comment (sebastian-aranda) [inline tests/e2e/mocked/plots/almanac.spec.js]: Adding a `UTC_TO_TAI_MS = 3700` constant would be good as this value is used in a couple other places.

**Review by sebastian-aranda — APPROVED** [review]
Great work! thank you very much @nemovc 🙌

File: Jenkinsfile [hunk 6]
```diff
@@ -36,15 +36,26 @@ pipeline {
...
         }
       }
     }
+    stage("Run e2e tests") {
+      steps {
+        script {
+          sh """
+            source /home/saluser/.setup_dev.sh
+            npx playwright install chromium
+            npm run test:e2e
```
Comment (sebastian-aranda) [inline Jenkinsfile]: Jenkins jobs seem to be running great 🚀 . One final suggestion 💡 : please add more verbosity to list the tests that have been run (and not only hony many have passed/failed). I think this would do the job:

```suggestion
            npm run test:e2e -- --reporter=list
```
